beprof/functions.py

```diff
@@ -22,3 +22,33 @@ def subtract(curve1, curve2, def_val=0):
     obj = curve1.__class__(np.stack((coord1, coord2), axis=1),
                            **curve1.__dict__['metadata'])
     return obj
+
+
+def medfilt(vector, window):
+    """
+    Apply a window-length median filter to a 1D array vector.
+
+    Should get rid of 'spike' value 15.
+    >>> print(medfilt(np.array([1., 15., 1., 1., 1.]), 3))
+    [ 1.  1.  1.  1.  1.]
+
+    The 'edge' case is a bit tricky...
+    >>> print(medfilt(np.array([15., 1., 1., 1., 1.]), 3))
+    [ 15.   1.   1.   1.   1.]
+    """
+    if not window % 2 == 1:
+        raise ValueError("Median filter length must be odd.")
+    if not vector.ndim == 1:
+        raise ValueError("Input must be one-dimensional.")
+
+    k = (window - 1) // 2  # window movement
+    result = np.zeros((len(vector), window), dtype=vector.dtype)
+    result[:, k] = vector
+    for i in range(k):
+        j = k - i
+        result[j:, i] = vector[:-j]
+        result[:j, i] = vector[0]
+        result[:-j, -(i + 1)] = vector[j:]
+        result[-j:, -(i + 1)] = vector[-1]
```

Comment:
I think main goal here is just to toss scipy away and see if it resolves other issues.